.github/workflows/main.yaml

```
@@ -0,0 +1,44 @@
+name: Build and push redis package
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-on-push:
+    if: github.event_name == 'push'
+    runs-on: ubuntu-latest
+    steps:     
+
+      - uses: actions/checkout@v3
+
+      - name: install spin
+        uses: fermyon/actions/spin/setup@v1
+        with:
+          github_token: ${{ github.token }}
+
+      - name: install go
+        uses: actions/setup-go@v3
```

The latest version is v5: https://github.com/actions/setup-go?tab=readme-ov-file#v5
```suggestion
        uses: actions/setup-go@v5
```